### PR TITLE
🐛 Fix helm and olm release workflows

### LINKED ISSUE #264: Investigate why OLM E2E fails during publishing
Seeing a failure during publishing a new version during the OLM E2E validation:

Runs locally just fine:
```
jdiaz@joels-air mondoo-operator (main *) % operator-sdk run bundle --namespace mondoo-operator --timeout 5m0s ghcr.io/mondoohq/mondoo-operator-bundle:v0.2.2
INFO[0013] Successfully created registry pod: ghcr-io-mondoohq-mondoo-operator-bundle-v0-2-2 
INFO[0013] Created CatalogSource: mondoo-operator-catalog 
INFO[0013] OperatorGroup "operator-sdk-og" created      
INFO[0013] Created Subscription: mondoo-operator-v0-2-2-sub 
INFO[0018] Approved InstallPlan install-74n8s for the Subscription: mondoo-operator-v0-2-2-sub 
INFO[0018] Waiting for ClusterServiceVersion "mondoo-operator/mondoo-operator.v0.2.2" to reach 'Succeeded' phase 
INFO[0018]   Waiting for ClusterServiceVersion "mondoo-operator/mondoo-operator.v0.2.2" to appear 
INFO[0026]   Found ClusterServiceVersion "mondoo-operator/mondoo-operator.v0.2.2" phase: Pending 
INFO[0028]   Found ClusterServiceVersion "mondoo-operator/mondoo-operator.v0.2.2" phase: Installing 
INFO[0039]   Found ClusterServiceVersion "mondoo-operator/mondoo-operator.v0.2.2" phase: Succeeded 
INFO[0039] OLM has successfully installed "mondoo-operator.v0.2.2" 
jdiaz@joels-air mondoo-operator (main *) % 
```

### PR DESCRIPTION
**Pull request by imilchev**
Fixes #264

Signed-off-by: Ivan Milchev <ivan@mondoo.com>

### CODE REVIEW

**Comment by imilchev** [comment]
The idea now is that the integration tests optionally deploy the operator. It is possible to say that the operator was deployed externally. If that is the case then the installation steps are skipped and only the actual tests are executed.

This allows us to actually test the functionality for the different installation types.

P.S.: if we want this to be idea we might consider even embedding the helm and operator SDK installation in the Go code so you get to run the steps locally as well. Future problem though...

**Review by joelddiaz — COMMENTED** [review]
I see helm now has the --wait, and the OLM setup still looks fundamentally the same.

What precisely is the thing that fixes what was failing in the Helm e2e, and what precisely is the thing that fixes what was failing in the OLM e2e?

**Comment by imilchev** [comment]
They were trying to call 'make test/deployment' which did not exist for a long time. That was the old e2e implementation.

The lack of wait was a bug too as this cause the gathering of logs to crash the entire workflow as well.

**Comment by joelddiaz** [comment]
> They were trying to call 'make test/deployment' which did not exist for a long time. That was the old e2e implementation.
> 
> The lack of wait was a bug too as this cause the gathering of logs to crash the entire workflow as well.

Oh, now that I'm looking at the most recent logs, it may be that the helm bits were fixed (where it was originally broken) because of https://github.com/mondoohq/mondoo-operator/pull/325 (where we now put the proper version strings in the generated helm charts). That fix would then expose the missing 'make test/deployment'.

Now how is the OLM e2e fixed with this PR? I see the same failure signature with the most recent run from earlier today https://github.com/mondoohq/mondoo-operator/runs/6373277232?check_suite_focus=true#step:8:77

**Comment by imilchev** [comment]
> > They were trying to call 'make test/deployment' which did not exist for a long time. That was the old e2e implementation.
> > The lack of wait was a bug too as this cause the gathering of logs to crash the entire workflow as well.
> 
> Oh, now that I'm looking at the most recent logs, it may be that the helm bits were fixed (where it was originally broken) because of #325 (where we now put the proper version strings in the generated helm charts). That fix would then expose the missing 'make test/deployment'.
> 
> Now how is the OLM e2e fixed with this PR? I see the same failure signature with the most recent run from earlier today https://github.com/mondoohq/mondoo-operator/runs/6373277232?check_suite_focus=true#step:8:77

I enabled the OLM workflow for this PR. Let's see if it will be successful

**Comment by imilchev** [comment]
> > They were trying to call 'make test/deployment' which did not exist for a long time. That was the old e2e implementation.
> > The lack of wait was a bug too as this cause the gathering of logs to crash the entire workflow as well.
> 
> Oh, now that I'm looking at the most recent logs, it may be that the helm bits were fixed (where it was originally broken) because of #325 (where we now put the proper version strings in the generated helm charts). That fix would then expose the missing 'make test/deployment'.
> 
> Now how is the OLM e2e fixed with this PR? I see the same failure signature with the most recent run from earlier today https://github.com/mondoohq/mondoo-operator/runs/6373277232?check_suite_focus=true#step:8:77

Apparently, in newer version of OLM there is some wait that is missing in the installation. Then we run the bundle before all the OLM pods are fully running and they get stuck. I added 30s sleep between the OLM install and the bundle run and it seems to be working fine now

**Comment by imilchev** [comment]
https://github.com/operator-framework/operator-lifecycle-manager/issues/2454

File: .github/workflows/e2e-olm.yaml [hunk 1]
```diff
@@ -29,43 +29,66 @@ jobs:
...
+      - name: Gather running pods
+        if: failure()
         run: |
-          curl -sSL http://mondoo.io/download.sh | bash
-          mv mondoo /usr/local/bin/ 
-          make test/deployment
+          kubectl get pods -A
+          mkdir -p tests/integration/_output
+          kubectl get pods -n mondoo-operator -o yaml >> tests/integration/_output/olm-installation-pods.log
+          kubectl logs -n mondoo-operator ghcr-io-mondoohq-mondoo-operator-bundle-v0-2-8 >> tests/integration/_output/olm-installation-pods.log
```
Comment (joelddiaz) [inline .github/workflows/e2e-olm.yaml:68]: It will not always be named this

File: .github/workflows/e2e-olm.yaml [hunk 2]
```diff
@@ -29,43 +29,66 @@ jobs:
...
           gpg -u "Operator SDK (release) <cncf-operator-sdk@cncf.io>" --verify checksums.txt.asc
           grep operator-sdk_${OS}_${ARCH} checksums.txt | sha256sum -c -
           chmod +x operator-sdk_${OS}_${ARCH} && sudo mv operator-sdk_${OS}_${ARCH} /usr/local/bin/operator-sdk
-      - name: Install olm
+
+      # Now that dependencies are cached the tests start almost immediately after minikube has started
+      # this makes tests fail occasionally. This sleep gives the runner some time to become more stable
+      # before the test execution starts.
+      - name: Wait a bit for the runner to become more stable
+        run: sleep 60
```
Comment (joelddiaz) [inline .github/workflows/e2e-olm.yaml:53]: So this was the missing magic step???  ¯\_(ツ)_/¯

File: .github/workflows/e2e-olm.yaml [hunk 3]
```diff
@@ -29,43 +29,66 @@ jobs:
...
+
+      - name: Install Mondoo operator using OLM
         run: |
-          trap 'echo "Saving OLM logs" ; kubectl get pods --namespace mondoo-operator -o yaml >> olm-logs.txt ; kubectl get pods -o name -n mondoo-operator | xargs -I{} kubectl logs --namespace mondoo-operator --prefix {} >> olm-logs.txt' EXIT
           operator-sdk olm install
+          sleep 30
```
Comment (imilchev) [inline .github/workflows/e2e-olm.yaml:58]: this fixes the stuck installation

**Review by joelddiaz — APPROVED** [review]
🥇

**Pull request merged**